### ISSUE TITLE
Turn into zenodo pub?

### ISSUE THREAD

**Issue by seltmann** [Big-Bee-Network/3bee.gif#1]
Hi @jhpoelen This is great! Can I go ahead and turn this into a zenodo pub?

**Comment by jhpoelen** [comment]
@seltmann glad you like the spinning bee! With some minor changes, the content would be suited for Zenodo pubs. The idea is to make sure that the data/... folder gets flattened into the flat file list. This way, content published into Zenodo can be indexed/accessed separately, greatly improving the re-use capabilities of the files. 

Why do you need it by? If you are in a hurry just holler and I'll try and do it today.

thx,
-jorrit

**Comment by jhpoelen** [comment]
planning to work on this in the next few hours. Please holler if you already made alternate plans.

**Comment by jhpoelen** [comment]
Thar she blows! https://en.wiktionary.org/wiki/thar_she_blows#English -

Big Bee Community, & Poelen, Jorrit H. (2022). UCSB-IZC0001219 - Bee Library - 73e389aa-5886-4c48-8778-ba8932d1bd7e hash://sha256/285af8cff9d15fe008193c53f04caa5bd0a14642523fb26e5bd94fe720cf0b78 hash://md5/82862398891da6f968688c61c783b5b6 (0.0.1) [Data set]. Zenodo. https://doi.org/10.5281/zenodo.7113719

**Comment by jhpoelen** [comment]
Note that this process can be automated when needed. I created this pub mostly automatically.

note that the processes (e.g. ,preston track, make.sh) as well as the programs (preston.jar, make.sh) are archived along with their relation to input/output digital assets. In other words, you can trace the sources of bee.gif to their origin including the programs used to do processing steps (e.g., tracking, generating derived gif image).

Also, note that because the data is kept in both Zenodo, Github and indexed by https://softwareheritage.org , you can clone the repo by running:

preston origin --remote https://linker.bio hash://sha256/285af8cff9d15fe008193c53f04caa5bd0a14642523fb26e5bd94fe720cf0b78 | preston cat | preston cat --remote https://linker.bio > /dev/null

and bee.gif can be found via:

https://linker.bio/hash://sha256/e1ce89711faa2a81e40f38560166da4ac36c51d17a467705e559d6b3f4c0bf31 (transparently retrieved via https://softwareheritage.org)

https://linker.bio/hash://md5/9fc984d691fced0a5e83d17a00557e4c (transparently retrieved via Zenodo.org) 

![image](https://user-images.githubusercontent.com/1084872/192321965-a11b0914-febc-4c78-b43b-e08516ee7b44.png)

![image](https://user-images.githubusercontent.com/1084872/192321872-c694f721-87b3-4243-8e52-d11d38ee9083.png)


@seltmann can you feel the excitement ?

**Comment by seltmann** [comment]
@jhpoelen very exciting and cool! I see this linking to issue https://github.com/bio-guoda/preston/issues/146

With that in mind, how would you see this gif linking to the models? Should those be added to this publication, or a separate publication and linked? I see them being added to this publication and the gif being a display of the images that make up the model. The model files are contained in the drive folder below.

https://drive.google.com/file/d/1HbSqyUvYtz43Lx403pKZ2q8c0Lf2d6dH/view?usp=sharing

**Comment by jhpoelen** [comment]
@seltmann I see the animated gif as a way to quickly see all the source images. Also, because of time restraints, I see the animated gif as a placeholder for the situation in which we can render the model nicely in the browser. 

If you'd like, I can add the model files to the publication. The main issue I have is how to render the model without have to install special programs, especially because I don't fully understand how the three files (obj, mtl, jpg) are linked.

Anyways, I'll add the obj, mtl, jpg to the pub in a minute. 

Thanks for being patient with me here.

**Comment by jhpoelen** [comment]
found a viewer - https://github.com/enjikaka/obj-viewer , working on building a little thing to help load the stuff.

**Comment by seltmann** [comment]
@jhpoelen I dont think it is necessary to use a viewer for the model. I see this requirement as being unnecessary and somewhat of a trap that gets people creating highly specialized and hard-to-support databases. We can use the .gif of the images that make up the model and screenshots of the complete model. 

Please change the title to include taxon name: Xylocopa sonorina and catalogNumber UCSB-IZC00012194 if possible.

Xylocopa varipuncta is a synonym and X. sonorina is the accepted name.

Also, please add me as a co-author. https://orcid.org/0000-0001-5354-6048

PNG of the model:
![UCSB-IZC00012194](https://user-images.githubusercontent.com/1044474/192341438-44daad72-8375-4005-bae5-4f2933dd2d31.png)

**Comment by jhpoelen** [comment]
@seltmann thanks for clarifying. Do you still have the pre-processed images that were derived from the originals? If so, please share them somehow, and I'll include them also. With this, I can say ( in a machine readable way: these images were used to build that model ) 

also, I'll update the title and add you as an author.

**Comment by seltmann** [comment]
@jhpoelen I wont be able to get to this today, but lets revisit after I return from my trip (October 5). I am hearing I need to clarify more the relationship between the model files and the images. There are several steps in the processing from the raw images to the models. This is all done in Agisoft Metashape and includes masking all of the images and aligning them. There is no pre-processing on the images before putting them in Metashape (in this case).

**Comment by jhpoelen** [comment]
Sounds good. 

I've just found a handy web-based 3d viewer that can load stuff straight from the files 

https://3dviewer.net/#model=https://raw.githubusercontent.com/Big-Bee-Network/3bee.gif/main/UCSB-IZC00012194.obj,https://raw.githubusercontent.com/Big-Bee-Network/3bee.gif/main/UCSB-IZC00012194.mtl,https://raw.githubusercontent.com/Big-Bee-Network/3bee.gif/main/UCSB-IZC00012194.jpg

**Comment by seltmann** [comment]
Nice!! thats great, so we can display the content in many formats! perfect

**Comment by jhpoelen** [comment]
hey just published a new zenodo pub, also included github readme with click to render in browser.

see https://github.com/big-bee-network/3bee.gif and associated zenodo page. 

Please confirm that this is what you were looking for.

**Comment by jhpoelen** [comment]
See also 

Big Bee Community, Poelen, Jorrit H., & Seltmann, Katja. (2022). Xylocopa sonorina - UCSB-IZC0001219 - Bee Library - 73e389aa-5886-4c48-8778-ba8932d1bd7e hash://sha256/96bfde1efa599e0e8e61de18b14d61dd308737f684950e4079c04e9bc0f33958 hash://md5/4940f68c84cffa4412f7ffb98bb255bd (0.0.2) [Data set]. Zenodo. https://doi.org/10.5281/zenodo.7114321
